modules/core/src/utils/math-utils.js

```diff
@@ -117,16 +117,34 @@ export function fp64LowPart(x) {
   return x - Math.fround(x);
 }
 
+let scratchArray;
+
 /**
- * Calculate the low part of a position
- * @param array {number} - the input position
- * @returns {array} - the lower 32 bit of the position
+ * Split a Float64Array into a double-length Float32Array
+ * @param typedArray {Float64Array}
+ * @param size {Number} - per attribute size
+ * @param [startIndex] {Number} - start index in the source array
+ * @param [endIndex] {Number} - end index in the source array
+ * @returns {Float32Array} - high part, low part for each attribute:
+    [1xHi, 1yHi, 1zHi, 1xLow, 1yLow, 1zLow, 2xHi, ...]
  */
-const target = new Array(2);
-export function positionFp64LowPart(array) {
-  target[0] = fp64LowPart(array[0]);
-  target[1] = fp64LowPart(array[1]);
-  // TODO - support 64-bit in z
-  // target[2] = fp64LowPart(array[2] || 0);
-  return target;
+export function toDoublePrecisionArray(typedArray, {size = 1, startIndex = 0, endIndex}) {
+  if (!Number.isFinite(endIndex)) {
+    endIndex = typedArray.length;
+  }
+  const count = (endIndex - startIndex) / size;
+  scratchArray = typedArrayManager.allocate(scratchArray, count, {
+    type: Float32Array,
+    size: size * 2
+  });
+
+  for (let sourceIndex = startIndex, targetIndex = 0; sourceIndex < endIndex; targetIndex += size) {
```

Comment:
Code-style Nit: Too compact, please separate this to individual rows. 
1. We never do multiple initializations on a single line. 
2. since we also increment targetIndex inside the loop body, I recommend we keep both increments in the loop body, or increment with `size + 1` in the loop header and avoid incrementing in the body.